src/OrchardCore.Modules/OrchardCore.Users/Workflows/Activities/RegisterUserTask.cs

```diff
@@ -0,0 +1,151 @@
+
+        // Returns the possible outcomes of this activity.
+        public override IEnumerable<Outcome> GetPossibleOutcomes(WorkflowExecutionContext workflowContext, ActivityContext activityContext)
+        {
+            return Outcomes(T["Done"], T["Valid"], T["Invalid"]);
+        }
+
+        // This is the heart of the activity and actually performs the work to be done.
+        public override async Task<ActivityExecutionResult> ExecuteAsync(WorkflowExecutionContext workflowContext, ActivityContext activityContext)
+        {
+            bool isValid = false;
+            IFormCollection form = null;
+            string email = null;
+            if (_httpContextAccessor.HttpContext != null)
+            {
+                form = _httpContextAccessor.HttpContext.Request.Form;
+                email = form["Email"];
+                isValid = !string.IsNullOrWhiteSpace(email);
+            }
+            var outcome = isValid ? "Valid" : "Invalid";
+
+
+            if (isValid)
+            {
+                var userName = form["UserName"];
+                if (string.IsNullOrWhiteSpace(userName))
+                    userName = email;
+
+                var errors = new Dictionary<string, string>();
+                var user = (User)await _userService.CreateUserAsync(new User() { UserName = userName, Email = email }, null, (key, message) => errors.Add(key, message));
+                if (errors.Count > 0)
+                {
+                    var updater = _updateModelAccessor.ModelUpdater;
+                    if (updater != null)
+                    {
+                        foreach (var item in errors)
+                        {
+                            updater.ModelState.TryAddModelError(item.Key, T[item.Value]);
+                        }
+                    }
+                    outcome = "Invalid";
+                }
+                else if (SendConfirmationEmail)
+                {
+                    var code = await _userManager.GenerateEmailConfirmationTokenAsync(user);
+                    var request = _httpContextAccessor.HttpContext.Request;
+                    var uriBuilder = new UriBuilder(request.Scheme, request.Host.Host);
+                    if (request.Host.Port.HasValue)
+                        uriBuilder.Port = request.Host.Port.Value;
+                    uriBuilder.Path = "OrchardCore.Users/Registration/ConfirmEmail";
+                    uriBuilder.Query = string.Format("userId={0}&code={1}", user.Id, code);
+                    workflowContext.Properties["EmailConfirmationUrl"] = uriBuilder.Uri.ToString();
+
+
+                    var body = await _expressionEvaluator.EvaluateAsync(ConfirmationEmailTemplate, workflowContext);
+                    var localized = new LocalizedHtmlString(nameof(RegisterUserTask), body);
+                    var message = new MailMessage() { Body = localized.IsResourceNotFound ? body : localized.Value, IsBodyHtml = true, BodyEncoding = Encoding.UTF8 };
```

Comment:
@MichaelPetrinolis Do you remember or understand what you were trying to do here with regards to localization? I don't understand what it would do better than just assign body directly?
/cc @hishamco if you have any idea also.
I could think that you are trying to allow for localizing the body, but I don't think it would work.

Reply:
Basically he checking the existing of the resource `RegisterUserTask` and set the body, but I don't understand why you need to do this check while the localization APIs did it for you moreover I don't think there a value of creating a localizer here, so I don't think it world work too ;)

Reply:
I'm asking myself how you checked a PR since merged few years ago ;)

Reply:
To tell the truth, it was one of my first PRs to learn OrchardCore. Watched the 'build a theme from scratch' demo and tried it my self :) Don't remember what drive me to this decision at that time.

Reply:
I can send a PR to remove it in order to avoid the confusion

Reply:
I already have it covered, thanks, found other places I want to clean up.

Reply:
I just want to clean it up, but while you already started it's in your plate